File: SDKs/googleapis.com/apikeys/v2/go/pkg/models/operations/apikeyskeyslookupkey.go
package operations

import (
"openapi/pkg/models/shared")

type ApikeysKeysLookupKeyQueryParams struct {
    DollarXgafv *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
    AccessToken *string `queryParam:"style=form,explode=true,name=access_token"`
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Callback *string `queryParam:"style=form,explode=true,name=callback"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    KeyString *string `queryParam:"style=form,explode=true,name=keyString"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    UploadType *string `queryParam:"style=form,explode=true,name=uploadType"`
    UploadProtocol *string `queryParam:"style=form,explode=true,name=upload_protocol"`
    
}

type ApikeysKeysLookupKeySecurityOption1 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type ApikeysKeysLookupKeySecurityOption2 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type ApikeysKeysLookupKeySecurity struct {
    Option1 *ApikeysKeysLookupKeySecurityOption1 `security:"option"`
    Option2 *ApikeysKeysLookupKeySecurityOption2 `security:"option"`
    
}

type ApikeysKeysLookupKeyRequest struct {
    QueryParams ApikeysKeysLookupKeyQueryParams 
    Security ApikeysKeysLookupKeySecurity 
    
}

type ApikeysKeysLookupKeyResponse struct {
    ContentType string 
    StatusCode int64 
    V2LookupKeyResponse *shared.V2LookupKeyResponse 
    
}

File: SDKs/googleapis.com/apikeys/v2/typescript/src/sdk/models/operations/apikeyskeyslookupkey.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class ApikeysKeysLookupKeyQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=keyString" })
  keyString?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @Metadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;
}


export class ApikeysKeysLookupKeySecurityOption1 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class ApikeysKeysLookupKeySecurityOption2 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class ApikeysKeysLookupKeySecurity extends SpeakeasyBase {
  @Metadata({ data: "security, option=true" })
  option1?: ApikeysKeysLookupKeySecurityOption1;

  @Metadata({ data: "security, option=true" })
  option2?: ApikeysKeysLookupKeySecurityOption2;
}


export class ApikeysKeysLookupKeyRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: ApikeysKeysLookupKeyQueryParams;

  @Metadata()
  security: ApikeysKeysLookupKeySecurity;
}


export class ApikeysKeysLookupKeyResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  v2LookupKeyResponse?: shared.V2LookupKeyResponse;
}
